attestation/secretscan/secretscan.go

```diff
@@ -0,0 +1,449 @@
+// Copyright 2025 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secretscan
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/in-toto/go-witness/attestation"
+	"github.com/in-toto/go-witness/cryptoutil"
+	"github.com/in-toto/go-witness/log"
+	"github.com/in-toto/go-witness/registry"
+	"github.com/invopop/jsonschema"
+	"github.com/zricethezav/gitleaks/v8/detect"
+)
+
+const (
+	Name    = "secretscan"
+	Type    = "https://witness.dev/attestations/secretscan/v0.1"
```

Comment:
There's going to be a race condition here with other post product attestors. If a SLSA attestor or SBOM detector have secrets, they may or may not be detected by this.

Reply:
lets not run it against the post product attestors, we are already scanning the products